File: Leetcode 75/Day 1 - Prefix Sum/1. Running Sum of 1D Array (1480)/RunningSumOf1DArray.java
class Solution {
    public int[] runningSum(int[] nums) {

        if (nums.length == 1) {
            return nums;
        }

        int curSum = nums[0];

        for (int i = 1; i < nums.length; i++) {
            nums[i] += curSum;
            curSum = nums[i];
        }

        return nums;
    }
}

File: Leetcode 75/Day 1 - Prefix Sum/1. Running Sum of 1D Array (1480)/RunningSumOf1DArray.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        curSum = nums[0]

        if len(nums) == 1:
            return curSum

        for i in range(1, len(nums)):

            nums[i] += curSum

            curSum = nums[i]

        return nums
